codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -669,12 +671,21 @@ public void operationComplete(ChannelFuture future) throws Exception {
     public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStreamId, final long errorCode,
                                 final ByteBuf debugData, ChannelPromise promise) {
         try {
+            promise = promise.unvoid();
             final Http2Connection connection = connection();
-            if (connection.goAwaySent() && lastStreamId > connection.remote().lastStreamKnownByPeer()) {
-                throw connectionError(PROTOCOL_ERROR, "Last stream identifier must not increase between " +
-                                                      "sending multiple GOAWAY frames (was '%d', is '%d').",
-                                      connection.remote().lastStreamKnownByPeer(), lastStreamId);
+            if (connection().goAwaySent()) {
```

Comment:
@buchgr does this really belong to this PR and is not an extra issue ? If so can you please open another issue so we know it was fixed.